CMakeLists.txt

```diff
@@ -318,18 +320,20 @@ add_custom_command (OUTPUT ${DIST_FILE}
       ${ALL_DIST} ${dist_generated_depends}
 )
 
-add_custom_target (dist DEPENDS ${DIST_FILE})
-
-add_custom_target (distcheck DEPENDS dist
-   COMMAND ${CMAKE_COMMAND}
-      -D CMAKE_MODULE_PATH=${PROJECT_SOURCE_DIR}/cmake/make_dist
-      -D CMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}
-      -D PACKAGE_PREFIX=${PACKAGE_PREFIX}
-      -D CMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
-      -P ${PROJECT_SOURCE_DIR}/cmake/make_dist/MakeDistCheck.cmake
-)
+if(NOT (TARGET bson_shared OR TARGET mongoc_shared))
```

Comment:
Are the dist targets skipped because the bundled C driver wouldn't get included, so rebuilding from the tarball would not necessarily succeed?

If a user opted out of bundling by setting `BUNDLED_MONGOC` to `OFF`, should the dist targets be added?

Also, I suggest adding something similar to the comment you added elsewhere noting "If these targets exist, then libmongoc has already been included as a project sub-directory" since that isn't clear at a glance.

Reply:
I have added a comment.

As far as the logic, my going in assumption was that someone wanting to bundle the C++ driver would also bundle the C driver and that in such a use case the `dist` and `distcheck` targets would be unused.

The reason for skipping the `dist` and `distcheck` targets is that the inclusion of the C driver via `add_subdirectory` will usually result in the creation of `dist` and `distcheck` targets.  Attempting to declare targets with those names again will result in an error.  The way I have implemented it a call to `add_subdirectory(mongo-c-driver)` will use the bundled C driver when building the C++ driver.  The user can disable the use of the bundled C driver by specifying `-DBUNDLED_MONGOC=OFF`, but the targets could still be declared in the current scope.  Since my assumption is that the targets would be unused/unnecessary, that simply skipping their creation was the simplest approach.

To be able to reliably allow for declaring the targets here based on `-DBUNDLED_MONGOC=OFF`, we would also need logic in the C driver that decides to skip creating the targets.  However, it is not clear to me how that would work in a simple and transparent way for the user.

The current implementation allows the user to do this:

```
add_subdirectory (mongo-c-driver)
add_subdirectory (mongo-cxx-driver)
```

Then the builds would work with default options.  Naturally, the user could specify non-default options, but those are entirely optional.  We could add an option so that the user could do something like this:

```
set(ENABLE_MONGOC_DIST OFF CACHE BOOL "" FORCE)
```

But that seems clunky and unnecessary (based on my initial assumption).  If my assumption is wrong then

Reply:
>  Since my assumption is that the targets would be unused/unnecessary, that simply skipping their creation was the simplest approach.

That seems like a fair assumption. I don't think users would want to produce a distribution tarball of the C/C++ driver if they are bundling.  I don't think a `ENABLE_MONGOC_DIST` is necessary.